### PR TITLE
fix(gomod): Don't append `indirect` mark twice

### LINKED ISSUE #20172: Go modules indirect updates creates multiple // indirect comments
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

3.0.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitLab Enterprise Edition 15.7.6-ee

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Suddenly we have started to get updates for indirect dependencies in Go. But the updates it suggest seems not quite correct. Example of changes in an merge request updating module github.com/go-stack/stack to v1.8.1: `github.com/go-stack/stack v1.8.1 // indirect // indirect`

<details>
<summary>Click to see screenshot of actual merge request</summary>

image

Another slightly different example:

image

</details>

Notice `// indirect // indirect`. I am seeing this in all merge/pull requests that Renovate creates for indirect dependencies (i.e. the presence of `// indirect // indirect` after the updated dependency). It should not be double up of `// indirect`.

This may be caused by:

https://github.com/renovatebot/renovate/blob/9e419616d025fb141cd2e69ca4519478c10c8ee8/lib/modules/manager/gomod/update.ts#L139-L141

Added January 15th in https://github.com/renovatebot/renovate/pull/19431

### Relevant debug logs

<details><summary>Logs</summary>

```json
{
  "name": "renovate",
  "hostname": "renovate-5bdfdb5cd9-4smsp",
  "pid": 8,
  "level": 20,
  "logContext": "i0mcVAyXo",
  "repository": "my-go-repository",
  "baseBranch": "main",
  "config": {
    "gomod": [
      {
        "packageFile": "go.mod",
        "deps": [
          {
            "managerData": {
              "lineNumber": 38,
              "multiLine": true
            },
            "depName": "github.com/go-stack/stack",
            "depType": "indirect",
            "currentValue": "v1.8.0",
            "datasource": "go",
            "enabled": false,
            "depIndex": 34,
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "v1.8.1",
                "newValue": "v1.8.1",
                "releaseTimestamp": "2021-08-18T18:48:21.000Z",
                "newMajor": 1,
                "newMinor": 8,
                "updateType": "patch",
                "branchName": "renovate-github.com-go-stack-stack-1.x"
              }
            ],
            "warnings": [],
            "versioning": "semver",
            "sourceUrl": "https://github.com/go-stack/stack",
            "currentVersion": "v1.8.0",
            "isSingleVersion": true,
            "fixedVersion": "v1.8.0"
          }
        ]
      }
    ]
  },
  "msg": "packageFiles with updates",
  "time": "2023-02-02T12:04:58.850Z",
  "v": 0
}
```

</details>

### Have you created a minimal reproduction repository?

https://github.com/mikaello/renovate-indirect-bug-repro, example of PR with the double `// indirect` comment: https://github.com/mikaello/renovate-indirect-bug-repro/pull/2/files

### PR DESCRIPTION
**Pull request by zharinov**
## Context

- Closes: #20172

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by zharinov** [comment]
But now I guess we also have to remove all the multiple occurrencess...

File: lib/modules/manager/gomod/update.ts [hunk 1]
```diff
@@ -137,7 +137,10 @@ export function updateDependency({
     }
 
     if (depType === 'indirect') {
-      newLine += ' // indirect';
+      newLine = newLine.replace(
+        regEx(/(?:\s*\/\/\s*indirect\s*)$/),
+        ' // indirect'
```
Comment (viceice) [inline lib/modules/manager/gomod/update.ts:142]: what if there's a `// indirect;` ? then we would append again?

**Review by viceice — COMMENTED** [review]
otherwise LGTM

File: lib/modules/manager/gomod/update.ts (hunk 1 above)
Comment (zharinov) [reply lib/modules/manager/gomod/update.ts:142]: Added to the regexp

**Comment by rarkins** [comment]
Strange that we didn't catch this in #19431 by @jamietanna

**Comment by zharinov** [comment]
Sometimes this happens, the `+=` operator looks is quite cool, so it's natural to attribute our own intention to it

File: lib/modules/manager/gomod/update.ts [hunk 2]
```diff
@@ -137,7 +137,10 @@ export function updateDependency({
     }
 
     if (depType === 'indirect') {
-      newLine += ' // indirect';
+      newLine = newLine.replace(
+        regEx(/\s*(?:\/\/\s*indirect(?:\s*;)?\s*)*$/),
```
Comment (Churro) [inline lib/modules/manager/gomod/update.ts:141]: Are you sure `\s` is the best choice here? It matches newlines too

**Review by rarkins — APPROVED** [review]

**Pull request merged**